sdk/test-utils/perfstress/src/recordingClient.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { HttpOperationResponse, URLBuilder } from "@azure/core-http";
+import { DefaultHttpClient, WebResource, WebResourceLike } from "@azure/core-http";
+
+const paths = {
+  playback: "/playback",
+  record: "/record",
+  start: "/start",
+  stop: "/stop"
+};
+
+export class RecordingHttpClient extends DefaultHttpClient {
```

Comment:
Bigger question.  Should we use a custom `HttpClient` to redirect to test-proxy, or can we use a `RequestPolicy` instead?  .NET uses policy.

Reply:
core v2 exposes the policies pipeline so I think it is possible for core v2. I am not sure about v1. cc @xirzec @jeremymeng

Reply:
Our clients don't allow adding policy addition at the user level, it stops at the SDK level.
Meaning.. services can add policies based on their needs but users don't have the same control. 
The only possible way right now is to go with httpClient.

I would love to move to the request policies way, and be at the same level as .NET and Java.
That would require the clients depending on core-v2 let users modify the policy addition by exposing an "addPolicy" method.

Reply:
@xirzec @chradek

Reply:
@HarshaNalluru you're right, the pipeline itself is not exposed but now that I think about it, I think exposing the pipeline is much more useful than exposing the client and I am not sure why we're not doing this but of course this out of the scope of this discussion.

Reply:
Hmm it's true that you can't add custom policies very easily in track1, but you could do something like write a simple HttpClient that applies a policy of your choice before calling into a cached version of the DefaultHttpClient ?

Reply:
@xirzec let's ignore core-v1 since we would eventually move to core-v2.
core-v1 services can use httpClient similar to what this PR has.

For core-v2, should we allow users to add policies into the pipeline? (Doing that would put us at the same level as .NET and Java.)

Reply:
We do this in some clients today, for example: https://github.com/Azure/azure-sdk-for-js/blob/ce2c1f5310bf0e71dbf8bc2b29d5731a9dfd961f/sdk/tables/data-tables/src/TableClient.ts#L268

Reply:
That's nice. 
Can we extend that for all the clients so that users have more control over the policies/pipeline?

Reply:
also, instead of worrying about updating the public surface for perf tests, we could instead access the private pipeline member directly using the `[]` notation, e.g. `client["pipeline"]`. It is not type-safe but it is not a big deal for tests.

Reply:
Resolving this.. will merge my branch that deals with the core-v2 SDKs.